server/app/views/applicant/TrustedIntermediaryDashboardView.java

```diff
@@ -369,11 +373,24 @@ private DivTag renderCardApplications(AccountModel account) {
     String programs =
         newestApplicantApplications.stream()
             .map(
-                application ->
-                    programRepository
-                        .getShallowProgramDefinition(application.getProgram())
+                application -> {
+                  try {
+                    return programService
```

Comment:
**suggestion (blocking):** hypothetically we don't need the full program definition, but this is this fixing the timeout issue because its getting the full program definition from the cache. Instead of using the full program definition, we can consider caching and using the shallow program definition (https://github.com/civiform/civiform/pull/6823/files). I'd hate for us to start using the full program definition as a workaround and then it would exacerbate the issue for CEs that don't have caching turned on (because that becomes even more calls without cache). Thoughts?

Reply:
Resolving since we chatted offline and this works when the cache is disabled too. Thanks @gwendolyngoetz!